ci/check_migration_files.sh

```diff
@@ -0,0 +1,29 @@
+#! /bin/bash
+
+set -euo pipefail
+# This file checks for the existence of downgrade scripts for every upgrade script that is changed in the branch.
+
+# list of migration files, and early exit if no migration scripts are touched
+migration_files=$(git diff --name-only origin/master | grep "src/backend/distributed/sql/citus--.*sql") || exit 0
+ret_value=0
+
+for file in $migration_files
+do
+    # There should always be 2 matches, and no need to avoid splitting here
+    # shellcheck disable=SC2207
+    versions=($(grep -Eo "\d\.\d-\d" <<< "$file"))
```

Comment:
Long flags are good to use in scripts, so commands become self documenting without having to read `man` pages.
```suggestion
    versions=($(grep --extended-regexp --only-matching "\d\.\d-\d" <<< "$file"))
```

Actually testing the regexp I found some issues with it though.
1. `\d` does not work with `--extended-regexp`/`-E`, you need `--perl`/`-P` for that.
2. To make it more future proof let's allow multiple digits in a row `\d+`
3. We might have `1.2.3` versions in the future instead of `1.2-3` versions, let's make it future proof against that.

So in the end I end up with this command:
```suggestion
    versions=($(grep --perl --only-matching "\d+\.\d+[-.]\d+" <<< "$file"))
```

This command actaully causes the script to always fail when a migration file has changed, because it cannot find the pattern.

Reply:
```
grep: unrecognized option `--perl'
usage: grep [-abcDEFGHhIiJLlmnOoqRSsUVvwxZ] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
```

BSD grep supports `\d` on `-E` extended regular expression mode. I will dig deeper into this, and if needed rewrite it using a separate cli tool that is more portable.

Reply:
Ah, that explains it. One easy way to fix it is to simply replace `\d` with `[0-9]`. That's portable everywhere, even without `-E` or `-P` I think. Also to be clear, if a long option does not exist on OSX, that's clearly a good reason to not use it.